### PR TITLE
Fix issue that the forbidden-apis check do not always work

### LINKED ISSUE #6355: It seems the forbidden apis check do not work
The build log:
```
Classes directory does not exist, forbidden apis check skipped
```

### PR DESCRIPTION
**Pull request by QiuMM**
Fix #6355.

### CODE REVIEW

**Comment by QiuMM** [comment]
I have modified the execution phase and run `mvn clean compile`, then forbidden apis check works fine, but I got this build error: 
```
[INFO] Scanning for classes to check...
[INFO] Reading bundled API signatures: jdk-unsafe-1.8
[INFO] Reading bundled API signatures: jdk-system-out
[INFO] Reading API signatures: /Users/qmm/work/github/incubator-druid/codestyle/joda-time-forbidden-apis.txt
[INFO] Reading API signatures: /Users/qmm/work/github/incubator-druid/codestyle/druid-forbidden-apis.txt
[INFO] Loading classes to check...
[INFO] Scanning classes for violations...
[ERROR] Forbidden field access: java.lang.System#out [prints to System.out; should only be used for debugging, not in production code]
[ERROR]   in org.apache.druid.cli.PullDependencies (PullDependencies.java:467)
[ERROR] Forbidden field access: java.lang.System#out [prints to System.out; should only be used for debugging, not in production code]
[ERROR]   in org.apache.druid.cli.validate.DruidJsonValidator (DruidJsonValidator.java:75)
[ERROR] Forbidden field access: java.lang.System#out [prints to System.out; should only be used for debugging, not in production code]
[ERROR]   in org.apache.druid.cli.Main (Main.java:119)
[ERROR] Forbidden field access: java.lang.System#out [prints to System.out; should only be used for debugging, not in production code]
[ERROR]   in org.apache.druid.cli.Main (Main.java:120)
[ERROR] Forbidden field access: java.lang.System#out [prints to System.out; should only be used for debugging, not in production code]
[ERROR]   in org.apache.druid.cli.Main (Main.java:121)
[ERROR] Forbidden field access: java.lang.System#out [prints to System.out; should only be used for debugging, not in production code]
[ERROR]   in org.apache.druid.cli.DumpSegment (DumpSegment.java:423)
[ERROR] Forbidden field access: java.lang.System#out [prints to System.out; should only be used for debugging, not in production code]
[ERROR]   in org.apache.druid.cli.Version (Version.java:36)
[ERROR] Scanned 74 (and 588 related) class file(s) for forbidden API invocations (in 0.34s), 7 error(s).
```

Some classes use `java.lang.System#out` and it's necessary, so I excluded these classes from forbidden apis check.

**Comment by QiuMM** [comment]
@leventov If there are compiled classes, forbidden-apis checks will work. That's the only reason. We can see in the CI `strict compilation`, the maven build command is `mvn clean -Pstrict -pl '!benchmarks' compile test-compile -B --fail-at-end`, `test-compile` is run after `compile`. So when maven run `test-compile` there are compiled classes for the forbidden-apis check plugin to work. And it just checks the source code classes not include test code classes because the test code classes have not been generated(before this pr, the `testCheck` Mojo runs in the `validate` phase, if change to `test-compile` phase, test code classes would also be checked).

**Comment by QiuMM** [comment]
And before this pr, some classes that use `java.lang.System#out` have not been checked out, e.g. `org.apache.druid.cli.Main`, which is flaky and it also confused me.

File: pom.xml [hunk 1]
```diff
@@ -934,12 +934,17 @@
                     <excludes>
                         <exclude>org/apache/druid/java/util/common/DateTimes$UtcFormatter.class</exclude>
                         <exclude>org/apache/druid/java/util/common/DateTimes.class</exclude>
+                        <exclude>org/apache/druid/cli/PullDependencies.class</exclude>
```
Comment (leventov) [inline pom.xml]: Instead of excluding whole files, could you annotate methods `@SuppressForbidden`

Comment (QiuMM) [reply pom.xml]: Nice suggestion. Done.

File: services/src/main/java/org/apache/druid/cli/validate/DruidJsonValidator.java [hunk 2]
```diff
@@ -69,6 +70,7 @@
     name = "validator",
     description = "Validates that a given Druid JSON object is correctly formatted"
 )
+@SuppressForbidden(reason = "System#out")
```
Comment (QiuMM) [inline services/src/main/java/org/apache/druid/cli/validate/DruidJsonValidator.java]: Place this annotation on the field like below not work, so I have to annotate the whole class.
```java
@SuppressForbidden(reason = "System#out")
private Writer logWriter = new PrintWriter(new OutputStreamWriter(System.out, StandardCharsets.UTF_8));
```

**Comment by jon-wei** [comment]
@QiuMM With this PR I am no longer able to build individual druid modules, e.g. druid-server:

```
cd server
mvn clean && mvn install -DskipTests=true
```

```
[INFO] ------------------------------------------------------------------------
[INFO] BUILD FAILURE
[INFO] ------------------------------------------------------------------------
[INFO] Total time: 15.322 s
[INFO] Finished at: 2018-09-26T16:42:00-07:00
[INFO] ------------------------------------------------------------------------
[ERROR] Failed to execute goal de.thetaphi:forbiddenapis:2.3:check (compile) on project druid-server: IO problem while reading files with API signatures.: /Users/jw/jw_repo/druid/server/codestyle/joda-time-forbidden-apis.txt (No such file or directory) -> [Help 1]
[ERROR] 
[ERROR] To see the full stack trace of the errors, re-run Maven with the -e switch.
[ERROR] Re-run Maven using the -X switch to enable full debug logging.
[ERROR] 
[ERROR] For more information about the errors and possible solutions, please read the following articles:
[ERROR] [Help 1] http://cwiki.apache.org/confluence/display/MAVEN/MojoExecutionException
```

**Pull request merged**